src/randomPortChromeConnection.ts

```diff
@@ -0,0 +1,53 @@
+import { chromeConnection } from 'vscode-chrome-debug-core';
+import { utils, chromeUtils } from "vscode-chrome-debug-core";
+import { logger } from "vscode-chrome-debug-core";
+import * as errors from './errors';
+
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+/**
+ * Chrome connection class that supports launching with --remote-debugging-port=0 to get a random port for the debug session
+ */
+export class RandomPortChromeConnection extends chromeConnection.ChromeConnection {
+
+    private userDataDir = undefined;
+    setUserDataDir(userDataDir: string) {
+        this.userDataDir = userDataDir;
+    }
+
+    /**
+     * Attach the websocket to the first available tab in the chrome instance with the given remote debugging port number.
+     * If we launched with port = 0, then this method will read the launched port from the user data directory, and wait until the port is open
+     * before calling super.attach
+     */
+    public attach(address = '127.0.0.1', port = 9222, targetUrl?: string, timeout = chromeConnection.ChromeConnection.ATTACH_TIMEOUT, extraCRDPChannelPort?: number): Promise<void> {
+        if(port === 0 && (this.userDataDir === undefined)) return errors.randomPortWithoutUserDataDir();
+        return utils.retryAsync(async () => {
+            const launchedPort = (port === 0 && this.userDataDir) ? await this.getLaunchedPort(address, this.userDataDir) : port;
```

Comment:
Will this work as espected for this.userDataDir === ""?

Reply:
Good point. Added a check for non-empty string.